tools/chorctl/cmd/repl_switch.go

```diff
@@ -0,0 +1,332 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	pb "github.com/clyso/chorus/proto/gen/go/chorus"
+	"github.com/clyso/chorus/tools/chorctl/internal/api"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func init() {
+	switchCmdInit()
+	switchZeroDowntimeCmdInit()
+	switchScheduledCmdInit()
+	switchDeleteCmdInit()
+}
+
+// get switch command:
+
+var (
+	switchCmdBucket string
+	switchCmdFrom   string
+	switchCmdTo     string
+	switchCmdUser   string
+	switchCmdWide   bool
+)
+
+func switchCmdInit() {
+	replCmd.AddCommand(switchCmd)
+	switchCmd.Flags().BoolVarP(&switchCmdWide, "wide", "w", false, "Wide output: shows switch status history")
+	switchCmd.Flags().StringVarP(&switchCmdBucket, "bucket", "b", "", "Filter by [B]ucket")
+	switchCmd.Flags().StringVarP(&switchCmdFrom, "from", "f", "", "Filter by [F]rom storage")
+	switchCmd.Flags().StringVarP(&switchCmdTo, "to", "t", "", "Filter by [T]o storage")
+	switchCmd.Flags().StringVarP(&switchCmdUser, "user", "u", "", "Filter by [U]ser")
+}
+
+var switchCmd = &cobra.Command{
+	Use:   "switch",
+	Short: "List all bucket replication switches",
+	Long: `Example:
+chorctl repl switch`,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		conn, err := api.Connect(ctx, address)
+		if err != nil {
+			logrus.WithError(err).Fatal("unable to connect to api")
+		}
+		defer conn.Close()
+
+		client := pb.NewChorusClient(conn)
+
+		resp, err := client.ListReplicationSwitches(ctx, &emptypb.Empty{})
+		api.PrintGrpcError(err)
+
+		w := tabwriter.NewWriter(os.Stdout, 10, 1, 5, ' ', 0)
+		fmt.Fprintln(w, "USER\tBUCKET\tFROM\tTO\tSTATUS\tLAST_STARTED\tDONE")
+		for _, switchStatus := range resp.Switches {
+			if switchCmdBucket != "" && switchStatus.ReplicationId.Bucket != switchCmdBucket {
+				continue
+			}
+			if switchCmdFrom != "" && switchStatus.ReplicationId.From != switchCmdFrom {
+				continue
+			}
+			if switchCmdTo != "" && switchStatus.ReplicationId.To != switchCmdTo {
+				continue
+			}
+			if switchCmdUser != "" && switchStatus.ReplicationId.User != switchCmdUser {
+				continue
+			}
```

Comment:
I'd probably move those for checks into separate function that would return `bool` and then do
```go
if shoudNotPrint := funcCall(...); shouldNotPrint {
    continue
}
```

Reply:
done